torch/optim/adamax.py

```diff
@@ -272,6 +276,13 @@ def _single_tensor_adamax(
         exp_avg = exp_avgs[i]
         exp_inf = exp_infs[i]
         step_t = state_steps[i]
+
+        # If compiling, the compiler will handle cudagraph checks, see note [torch.compile x capturable]
```

Comment:
Is it a problem to keep them here as well? It sounds like pretty trivial code for the compiler to handle?

Reply:
cc @mlazos

Reply:
We have these everywhere now--it may be time to look into whether we still need the is_compiling, but I think we can address it apart from this PR.

Reply:
Basically this is here because the compiler will decide whether to run cudagraphs or not, regardless of specifying capturable/not capturable. If some params aren't on cuda, we won't run cudagraphs and print a warning.